api/src/deployment.rs

```diff
@@ -49,10 +49,15 @@ impl Deployment {
         }
     }
 
-    fn from_bytes(fqdn: &str, project: ProjectName, crate_bytes: Vec<u8>) -> Self {
+    fn from_bytes(
+        fqdn: &str,
+        project: ProjectName,
+        crate_bytes: Vec<u8>,
+        initial_secrets: InitialSecrets,
```

Comment:
Won't the `Secrets.toml` file also be in `crate_bytes`? Could we possibly read them from there on the `api` end?